modules/core/arrow-test/src/main/kotlin/arrow/test/laws/ApplyLaws.kt

```diff
@@ -0,0 +1,63 @@
+package arrow.test.laws
+
+import arrow.Kind
+import arrow.test.generators.applicative
+import arrow.test.generators.functionAToB
+import arrow.test.generators.intSmall
+import arrow.typeclasses.Applicative
+import arrow.typeclasses.Apply
+import arrow.typeclasses.Eq
+import io.kotlintest.properties.Gen
+import io.kotlintest.properties.forAll
+
+object ApplyLaws {
+
+    fun <F, A> laws(AP: Applicative<F>, gen: Gen<A>, EQ: Eq<Kind<F, A>>): List<Law> =
```

Comment:
We should keep the laws separated between those that use Apply and those that use Applicative.

Then I realised all of them needed both lol

I'd keep them as ApplicativeLaws tbh.

Reply:
But do they? If just is only used to lift into `Kind<F, A>` then it might be better to either use a constructor function or a `Gen<Kind<F, A>>` directly. And all laws that strictly need just belong to applicative anyway. Although I am not too familiar with applicative laws, so idk how many that leaves o.O

Reply:
This is an issue we have in more places.

We should indeed be using `Gen<Kind<F, A>>` directly here. There is some low hanging fruit in the test setup.

Reply:
Hi all, I commented one interesting point about the laws in the PR description:

> It was weird to see how the laws already described for Aplicative doesn't match with the one I've found in other implementations like Cat's Apply or Scalaz's Apply typeclasses.

> I've updated the AplicativeLaws implementation to be divided in ApplyLaws and ApplyCartesianLaws. This change let us write some of our laws without coupling the implementation to any concrete type. I'm afraid I couldn't implement this for the cartesian laws because of the internal map implemented. Ideally, we'd like to don't even couple this implementation to the Applicative instance passed as parameter, but there is no way to lift any instance into Kind<F, A> w/o an Applicative instance. I've reviewed the laws implementation with @delr3ves thinking about how could reduce the coupling but it wasn't possible for this scenario.

Finally, as ``FunctorLaws`` need some dependencies I can't fix for now without rewriting most of the laws and this issue was supposed to bee "noob friendly" I decided not to refactor them and just move them to the ``ApplyLaw`` folder.

Are we keep the laws in the ``ApplicativeLaw`` folder and remove this part of the PR? I tried to use Gen<Kind<F, A>>, and a method to lift any value into Kind, but it is not so easy without refactoring the rest of the code laws.